logstreams/src/main/java/io/zeebe/logstreams/log/BufferedLogStreamReader.java

```diff
@@ -109,12 +108,40 @@ public void seekToFirstEvent() {
     seek(FIRST_POSITION);
   }
 
-  @Override
-  public void seekToLastEvent() {
-    seek(getLastPosition());
+  public long seekToEnd() {
+    // invalidate events first as the buffer content may change
+    invalidateBufferAndOffsets();
 
-    if (isNextEventInitialized()) {
-      checkIfNextEventIsCommitted();
+    final long blockAddress = logStorage.getFirstBlockAddress();
+    if (blockAddress < 0) {
+      // no block found => empty log
+      state = IteratorState.EMPTY_LOG_STREAM;
+      return -1;
+    } else {
+      readLastBlockIntoBuffer();
+      return completeEventsInBlockProcessor.getLastReadEventPosition();
+    }
+  }
+
+  private void readLastBlockIntoBuffer() {
```

Comment:
This method is really similar to 'readBlockIntoBuffer', is it possible to refactor this? I get that it's not efficient to find the last block and then reuse readBlockIntoBuffer because we'd be re-reading the same block but maybe we can still refactor this. What do you think?